### PR TITLE
add usename and password to eventbus mqtt configuration

### LINKED ISSUE #3035: Why not make eventbus client options configurable?
image

**What would you like to be added/modified**:

Make client ID, username, password  configurable.

**Why is this needed**:

In consideration of security, we usually need the mqtt broker to authenticate all the device client including eventbus.

### PR DESCRIPTION
**Pull request by gy95**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
add mqtt configuration: Username and Password to enhance security.
**Which issue(s) this PR fixes**:

Fixes #3035

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: edge/pkg/eventbus/mqtt/client.go [hunk 1]
```diff
@@ -58,9 +58,11 @@ var (
...
 type Client struct {
-	MQTTUrl string
-	PubCli  MQTT.Client
-	SubCli  MQTT.Client
+	MQTTUrl  string
+	Username string
+	Password string
+	PubCli   MQTT.Client
+	SubCli   MQTT.Client
 }
```
Comment (zc2638) [inline edge/pkg/eventbus/mqtt/client.go:66]: Do you consider supporting custom configuration of `clientID`? 
There may be a demand?

Comment (gy95) [reply edge/pkg/eventbus/mqtt/client.go:66]: clientID is generated by code like this, do we need to make it configurable?
https://github.com/kubeedge/kubeedge/blob/ea4dc19408cf6b424632b93a33c1aa52405b6090/edge/pkg/eventbus/mqtt/client.go#L143-L144

**Comment by fisherxu** [comment]
cc @sdghchj Please take a look :)

**Comment by sdghchj** [comment]
Thank you for your help to achieve it.

**Comment by gy95** [comment]
kindly ping @fisherxu  😄

File: edge/pkg/eventbus/mqtt/client.go (hunk 1 above)
Comment (gy95) [reply edge/pkg/eventbus/mqtt/client.go]: > 
done
add clientIDs for subClient and pubClient

File: pkg/apis/componentconfig/edgecore/v1alpha1/types.go [hunk 2]
```diff
@@ -336,6 +336,18 @@ type EventBus struct {
 	// MqttServerExternal indicates external mqtt broker url
 	// default "tcp://127.0.0.1:1883"
 	MqttServerExternal string `json:"mqttServerExternal,omitempty"`
+	// MqttSubClientID indicates mqtt subscribe ClientID
+	// default ""
+	MqttSubClientID string `json:"mqttSubClientID"`
```
Comment (fisherxu) [inline pkg/apis/componentconfig/edgecore/v1alpha1/types.go:341]:
```suggestion
	MqttSubClientID string `json:"mqttSubClientID,omitempty"`
```
Same as below.

Comment (gy95) [reply pkg/apis/componentconfig/edgecore/v1alpha1/types.go]: if add omitempty, if we use `edgecore --defaultconfig` cmd to generate config file, these fields will be empty, and just contain below things.
```
eventBus:
    enable: true
    eventBusTLS:
      enable: false
      tlsMqttCAFile: /etc/kubeedge/ca/rootCA.crt
      tlsMqttCertFile: /etc/kubeedge/certs/server.crt
      tlsMqttPrivateKeyFile: /etc/kubeedge/certs/server.key
    mqttMode: 2
    mqttQOS: 0
    mqttRetain: false
    mqttServerExternal: tcp://127.0.0.1:1883
    mqttServerInternal: tcp://127.0.0.1:1884
    mqttSessionQueueSize: 100
```

Comment (gy95) [reply pkg/apis/componentconfig/edgecore/v1alpha1/types.go]: seems not obvious for users to set username and password

Comment (fisherxu) [reply pkg/apis/componentconfig/edgecore/v1alpha1/types.go]: > if add omitempty, if we use edgecore --defaultconfig cmd to generate config file, these fields will be empty,

Make sense, we need ensure them printed in default config and omit in mincaonfig.

Comment (gy95) [reply pkg/apis/componentconfig/edgecore/v1alpha1/types.go:341]: even these fields are empty string, also printed
```
eventBus:
    enable: true
    eventBusTLS:
      enable: false
      tlsMqttCAFile: /etc/kubeedge/ca/rootCA.crt
      tlsMqttCertFile: /etc/kubeedge/certs/server.crt
      tlsMqttPrivateKeyFile: /etc/kubeedge/certs/server.key
    mqttMode: 2
    mqttPassword: ""
    mqttPubClientID: ""
    mqttQOS: 0
    mqttRetain: false
    mqttServerExternal: tcp://127.0.0.1:1883
    mqttServerInternal: tcp://127.0.0.1:1884
    mqttSessionQueueSize: 100
    mqttSubClientID: ""
    mqttUsername: ""
```

**Review by fisherxu — COMMENTED** [review]
/approve
/assign @zc2638 for final review.

**Pull request merged**